teamscale_client/client.py

```diff
@@ -634,6 +645,36 @@ def get_findings(self, uniform_path, timestamp, recursive=True):
             raise ServiceError("ERROR: GET {url}: {r.status_code}:{r.text}".format(url=service_url, r=response))
         return self._findings_from_json(response.json())
 
+    def get_finding_by_id(self, finding_id, branch=None, timestamp=None):
+        """Retrieves the finding with the given id.
+
+        Args:
+            finding_id (str): The id of the finding to retrieve.
+            branch (str): The branch from which the finding should be retrieved.
+            timestamp (datetime.datetime): The timestamp (unix format) for which to receive the finding.
+
+        Returns:
+             data.Finding: The retrieved finding.
+
+        Raises:
+            ServiceError: If anything goes wrong
+        """
+        service_url = self.get_project_service_url("findings-by-id") + finding_id
+
+        branch_stash = self.branch
+        self.branch = branch
+
+        parameters = {
+            "t": self._get_timestamp_parameter(timestamp=timestamp) if timestamp else 'HEAD',
```

Comment:
Can we add an optional parameter to `_get_timestamp_parameter` for passing a branch, that will be used if given, instead of the `self.branch` variable?

Reply:
Good idea